### PR TITLE
Repo: Rework README and create gitattributes

### PR DESCRIPTION
**Pull request by l-zeuch**
This pull request reworks the README and adds git attributes.

The README was quite outdated and fairly cluttered. This rewrite aims to make it easier to navigate and read. Important sections with the intention to "sell" this bot have been moved to the top to make them easier spot. Markdown syntax has been updated to be more consistent throughout the file, so that reading the raw document is becoming easier as well.
Some sections have been removed, as they are no longer true or necessary. Additionally, the step for building has been changed from `go build` to instead run the `build.sh` script to make an actually deployed version of the bot -- this has been mentioned several times to new users in `#self-hosting-discussion` on the server, so including that in the README is quite fair.

Git attributes have been introduced in an attempt to normalize line endings for certain files, e.g. shell vs. batch scripts -- these scripts can/will break if they have the wrong line ending (LF (*nix) vs. CRLF (Windows)). Moreover, because the vendored directory has a different name (`vendorr`), we tell linguist -- GitHub's language savant responsible for language statistics -- to exclude these directories from the statistic. Admittedly, a small detail, however I believe it is important, seeing that YAGPDB is in Go rather than JavaScript.

----
Seeing as this is a pull request of not small significance, please take your time to review this request as carefully as possible and do not hesitate to request changes.

Thanks in advance!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,129 +55,82 @@ via the Control Panel.
...
 ```
 
-After this, unless you wanna run it in testing mode using `YAGPDB_TESTING=yes` you have to run `cmd/yagpdb/copytemplates.sh` to copy all the plugin specific template files into the `cmd/yagpdb/templates/plugins` folder.
+### Hosting Standalone
 
-You can now run `./yagpdb`
+#### Requirements
 
-Configuration is done through environment variables. See `cmd/yagpdb/sampleenvfile` for what environment variables are available.
+* Recent GoLang (1.16 and newer is recommended)
```
Comment (jo3-l) [inline README.md:95]:
```suggestion
* Golang 1.16 or above
```
The go.mod requires 1.16,

File: README.md [hunk 2]
```diff
@@ -1,34 +1,52 @@
...
+* [Useful Links](#Useful-Links)
+* [Selfhosting](#Selfhosting)
+  - [General Setup](#General-Bot-Setup)
+  - [Dockerized](#Hosting-Dockerized)
+  - [Standalone](#Hosting-Standalone)
+    + [Requirements](#Requirements)
+    + [Setting Up](#Setting-Up)
+* [Databases](#Databases)
+  - [Updating](#Updating)
+* [Contributing](#Contributing)
```
[the remark this answers is not in the document]
Comment (l-zeuch) [reply README.md:20]: New to me... Let me know if we should keep it anyways, I don't mind either

[the remark this answers is not in the document]
Comment (3kh0) [reply README.md:20]: it would still be nice to have

**Review by 3kh0 — APPROVED** [review]
Looks good!

**Comment by l-zeuch** [comment]
Hi @nitinr-bst !

I'm wondering why this has been closed without comment, though you've changed the base branch -- which shows me you kind-of planned to merge this? If it's because of merge conflicts, that's not a problem, I'd resolve them.

**Comment by nitinr-bst** [comment]
> Hi @nitinr-bst !
> 
> I'm wondering why this has been closed without comment, though you've changed the base branch -- which shows me you kind-of planned to merge this? If it's because of merge conflicts, that's not a problem, I'd resolve them.

Pls resolve conflicts.

**Comment by l-zeuch** [comment]
> Pls resolve conflicts.
Done as we speak!
Thanks :+1:

**Pull request merged**